uftrace.c

```diff
@@ -1003,6 +1009,36 @@ static void free_opts(struct opts *opts)
 }
 
 #ifndef UNIT_TEST
+static void apply_default_opts(int *argc, char ***argv, struct opts *opts)
+{
+	char *basename = "default.opts";
+	char opts_file[PATH_MAX];
+	struct stat stbuf;
+
+	/* default.opts is only for analysis commands */
+	if (opts->mode == UFTRACE_MODE_RECORD ||
+	    opts->mode == UFTRACE_MODE_LIVE ||
+	    opts->mode == UFTRACE_MODE_RECV)
+		return;
+
+	/* this is not to override user given time-filter by default opts */
+	parsing_default_opts = true;
+
+	snprintf(opts_file, PATH_MAX, "%s/%s", opts->dirname, basename);
+	if (!stat(opts_file, &stbuf) && stbuf.st_size > 0) {
+		pr_dbg("apply '%s' option file\n", opts_file);
+		parse_opt_file(argc, argv, opts_file, opts);
+	}
+	else if (!strcmp(opts->dirname, UFTRACE_DIR_NAME) &&
+		 !access("./info", F_OK)) {
+		/* try again applying default.opts in the current dir */
+		if (!stat(basename, &stbuf) && stbuf.st_size > 0) {
+			pr_dbg("apply './%s' option file\n", basename);
+			parse_opt_file(argc, argv, basename, opts);
+		}
+	}
+}
+
```

Comment:
I think it's more useful if we can support any option in the `default.opts` file.  But let's think about it later..

Reply:
The `default.opts` can manually be modified to add more options later. I plan to use it when distributing some pre-recorded data to apply other default filters.

Reply:
But I'm afraid of the priority.  It's not acceptable to apply all the options like you did for the `-t` option.

Reply:
I also think that it might not be a perfect solution, but I think at least normal users will not modify `default.opts` manually. I will think about the better solution later.